pkg/plugin/plugin.go

```diff
@@ -20,53 +24,58 @@ type TLSFiles struct {
 	CLIENT    string `json:"client"`
 }
 
-func readTlsFile() (TLSFiles, error){
+func readTlsFile() (TLSFiles, error) {
 	home, err := os.UserHomeDir()
 	if err != nil {
-	  return TLSFiles{}, err
+		return TLSFiles{}, err
 	}
 	clientKey, err := os.ReadFile(home + "/.lightning/testnet/client-key.pem")
 	ca, err := os.ReadFile(home + ".lightning/testnet/ca.pem")
 	client, err := os.ReadFile(home + ".lightning/testnet/client.pem")
-	if err != nil{
-	  return TLSFiles{}, err
+	if err != nil {
+		return TLSFiles{}, err
 	}
 	files := TLSFiles{CA: string(ca), CLIENTKEY: string(clientKey), CLIENT: string(client)}
 	return files, nil
 }
 
-func startServer() {
-	r := gin.Default()
-	r.GET("/ping", func(c *gin.Context) {
+type StartServer[T PluginState] struct{}
+
+func (instance *StartServer[T]) Call(plugin *plugin.Plugin[PluginState], request map[string]string) (map[string]string, error) {
+	if plugin.State.Server != nil {
+		return map[string]string{"message": "Server is already running"}, nil
+	}
+	plugin.State.Server = gin.Default()
+
+	plugin.State.Server.GET("/ping", func(c *gin.Context) {
 		c.JSON(http.StatusOK, gin.H{
 			"message": "pong",
 		})
 	})
 
-	r.POST("/tls", func(c *gin.Context) {
+	plugin.State.Server.POST("/tls", func(c *gin.Context) {
 		var login LOGIN
 		c.BindJSON(&login)
-		if os.Getenv("password") != login.PASSWORD {
+		if plugin.State.Password != login.PASSWORD {
 			c.JSON(200, gin.H{"status": "Access Denied"})
 		}
```

Comment:
```suggestion
		if plugin.State.Password != login.PASSWORD {
			c.JSON(403, gin.H{"status": "Access Denied"})
			return
		}
```

We should stop the execution in case of error right? or I'm missing something?

In addition, the error should be 403 according with [wikipedia](https://en.wikipedia.org/wiki/HTTP_403#:~:text=HTTP%20403%20is%20an%20HTTP,but%20will%20not%20fulfill%20it.) it is no great as docs but the error is pretty common